### PR TITLE
Add ExtractDomain method for the identity/v3/tokens

### PR DESCRIPTION
**Pull request by outcatcher**
Mirrors https://github.com/gophercloud/gophercloud/pull/1712

OTC docs:
https://docs.otc.t-systems.com/en-us/api/iam/en-us_topic_0057845585.html

### CODE REVIEW

File: openstack/identity/v3/tokens/testing/fixtures.go [hunk 1]
```diff
@@ -209,3 +301,19 @@ func getGetResult(t *testing.T) tokens.GetResult {
...
 	return result
 }
+
+// ExpectedDomain contains expected domain extracted from token response.
+var ExpectedDomain = tokens.Domain{
+	ID:   "default",
+	Name: "Default",
+}
+
+func getGetDomainResult(t *testing.T) tokens.GetResult {
```
Comment (lego963) [inline openstack/identity/v3/tokens/testing/fixtures.go:311]: Misspell `getGet`

**Review by lego963 — CHANGES_REQUESTED** [review]
rename func name

File: openstack/identity/v3/tokens/testing/fixtures.go (hunk 1 above)
Comment (outcatcher) [reply openstack/identity/v3/tokens/testing/fixtures.go:311]: No, this is `Get "get domain"` result

File: openstack/identity/v3/tokens/testing/fixtures.go [hunk 2]
```diff
@@ -110,6 +110,98 @@ const TokenOutput = `
...
+      "password"
+    ],
+    "roles":[
+       {
+          "id":"434426788d5a451faf763b0e6db5aefb",
+          "name":"admin"
+       }
+    ],
+    "expires_at": "2019-09-18T23:12:32.000000Z",
+    "catalog":[
```
Comment (lego963) [inline openstack/identity/v3/tokens/testing/fixtures.go:130]: did you forget about ?
```
    "type": "identity",
    "id": "1331e5cff2a74d76b03da1225910e31d",
    "name": "iam",
```

**Review by lego963 — APPROVED** [review]

**Pull request merged**